### DISCUSSION TITLE
Impersonate action on EditRecord page - not logging in

### DISCUSSION THREAD

**Discussion by studiowizjo** [filamentphp/filament#1663] [Help]
As in title: I`m trying to add impersonate button action on EditRecord page:

```php
protected function getActions(): array
{
	return array_merge(parent::getActions(), [
		ButtonAction::make('impersonate')
			->label('Zaloguj jako')
			->action('impersonate'),
	]);
}

public function impersonate()
{
	auth()->loginUsingId($this->record->id, true);
	//also tried this, but it is the same: Filament::auth()->loginUsingId($this->record->id, true);
	return redirect()->back();
}
```

After clicking button, nothing happens, but when I refresh page I got error **Route [login] not defined.**.

This can be easily solved by adding to **routes/web.php**:

```php
Route::any('/login', Filament\Http\Livewire\Auth\Login::class)->name('login');
```

but the problem stays unsolved. For some reason user is not logged to given account. 

Auth guard is default.

**Comment by danharrin** [comment]
Have you tried redirecting to a specific route in your app?

**Comment by studiowizjo** [comment]
Nothing changed:

```php
public function impersonate()
{
	Filament::auth()->loginUsingId($this->record->id, true);
	return redirect()->route('filament.resources.users.index');
}
```

After click it logs out and redirects to login page.

**Reply by hrvoj3e** [reply]
Did you solve this?

I need something similar. I have a valid session in another app and would like to login user in filament admin via some special route so user won't have to login twice.
Do you have any suggestions?

**Reply by studiowizjo** [reply]
Unfortunately not. Instead I used this plugin - https://filamentphp.com/plugins/impersonate and temporary don`t have enought time to inspect how author solved it.
